cmd/commands/roles.go

```diff
@@ -81,7 +104,26 @@ func makeMemberCmdF(command *cobra.Command, args []string) error {
 			return errors.New("Unable to find user '" + args[i] + "'")
 		}
 
-		if _, err := a.UpdateUserRoles(user.Id, "system_user", true); err != nil {
+		systemUser := false
+		var newRoles []string
+
+		roles := strings.Fields(user.Roles)
+		for _, role := range roles {
+			switch role {
+			case model.SYSTEM_ADMIN_ROLE_ID:
+			default:
+				if role == model.SYSTEM_USER_ROLE_ID {
+					systemUser = true
+				}
+				newRoles = append(newRoles, role)
+			}
+		}
+
+		if !systemUser {
+			newRoles = append(roles, model.SYSTEM_USER_ROLE_ID)
```

Comment:
I don't know about this. The command help says it removes the system_admin permissions, but it doesn't say anything about the system_user at all. If the system_user role is mandatory, maybe must be solved in other place. If the system_user roles is not mandatory, it is adding the system_user role when the user have no reason to expect that.

Reply:
I think it's a bit subjective based on the docs. If you take it very literally, then yes, it should not add the `system_user` role if it is absent. However, in practice, I don't think that is what the user expects from this - they expect it to make the user an ordinary member rather than a system admin, which is what this behaviour (and the original behaviour it replaced) achieves.

Reply:
Makes sense :+1: